src/main/java/language/LanguageCommand.java

```diff
@@ -0,0 +1,383 @@
+                );
+    }
+
+    private EmbedBuilder handleCalculate(User user, InteractionBase event, SlashCommandInteractionOption option) {
+        Optional<Language> targetLanguageOpt = option.getOptionStringValueByName("target")
+                .flatMap(languageLogic::getByName);
+
+        if (targetLanguageOpt.isPresent()) {
+            Map<String, Try<Collection<Language>>> userLanguages = Maps.newHashMap();
+
+            option.getOptionStringValueByName("languages")
+                    .map(languages -> Arrays.stream(languages.split(","))
+                            .map(languageLogic::getByName)
+                            .flatMap(value -> value.map(Stream::of).orElse(Stream.empty()))
+                            .collect(Collectors.toSet())
+                    ).ifPresent(languages -> userLanguages.put("Entered", Try.success(languages)));
+
+            option.getOptionMentionableValueByName("characters")
+                    .map(PlayerHandler::getPlayersFromMentionable)
+                    .map(players -> players.stream()
+                            .map(userPlayerPair -> {
+                                Try<Collection<Language>> languageNames = SheetsHandler.getLanguages(userPlayerPair.getRight())
+                                        .map(languages -> languages.stream()
+                                                .map(languageLogic::getByName)
+                                                .flatMap(value -> value.map(Stream::of).orElse(Stream.empty()))
+                                                .collect(Collectors.toSet())
+                                        );
+                                return Pair.of(DiscordHelper.getUsernameFromInteraction(event, user), languageNames);
+                            })
+                    ).ifPresent(stream -> stream.forEach(
+                            pair -> userLanguages.put(pair.getLeft(), pair.getRight())
+                    ));
+
+            if (userLanguages.isEmpty()) {
+                PlayerHandler.getPlayersFromMentionable(user)
+                        .forEach(userPlayerPair -> userLanguages.put(
+                                DiscordHelper.getUsernameFromInteraction(event, user),
+                                SheetsHandler.getLanguages(userPlayerPair.getRight())
+                                        .map(languages -> languages.stream()
+                                                .map(languageLogic::getByName)
+                                                .flatMap(value -> value.map(Stream::of).orElse(Stream.empty()))
+                                                .collect(Collectors.toCollection(HashSet::new))
+                                        )
+                        ));
+            }
+
+            if (userLanguages.isEmpty()) {
+                return new EmbedBuilder()
+                        .setTitle("Failed to Calculate Linguistics Checks")
```

Comment:
Can you slap on a random color and call setAuthor for these embeds? There's a util class that can add a random color.